automation-tests/test/utilities.ts

```diff
@@ -0,0 +1,337 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { ChildProcess, exec } from 'child_process';
+import clipboard from 'clipboardy'
+import os from 'os';
+import vscode from 'vscode';
+import vscodeType from 'vscode';
+import {
+  DefaultTreeItem,
+  InputBox,
+  OutputView,
+  QuickOpenBox,
+  sleep,
+  StatusBar,
+  TerminalView,
+  TreeItem,
+  ViewItem,
+  Workbench
+} from 'wdio-vscode-service';
+import { CMD_KEY } from 'wdio-vscode-service/dist/constants';
+import {
+  EnvironmentSettings
+} from './environmentSettings';
+
+
+async function openFile(vscode: typeof vscodeType, file: string): Promise<vscode.TextEditor> {
+  const filePath = vscode.Uri.parse(file).fsPath;
+  const textDocument = await vscode.workspace.openTextDocument(filePath);
+  const textEditor = await vscode.window.showTextDocument(textDocument);
+
+  return textEditor;
+}
+
+async function openFolder(vscode: typeof vscodeType, folderPath: string): Promise<unknown> {
+  const folderPathUri = vscode.Uri.parse(folderPath);
+  const result = await vscode.commands.executeCommand('vscode.openFolder', folderPathUri);
+
+  // TODO: verify return type
+
+  return result;
+}
+
+function createFolder(folderPath: string): ChildProcess {
+  const childProcess = exec(`mkdir "${folderPath}"`);
+
+  return childProcess;
+}
+
+function removeFolder(folderPath: string): ChildProcess {
+  const childProcess = exec(`rm -rf "${folderPath}"`);
+
+  return childProcess;
+}
+
+async function pause(duration: number): Promise<void> {
+  await sleep(duration * EnvironmentSettings.getInstance().throttleFactor * 1000);
+}
+
+function log(message: string) {
+  console.log(message);
+}
+
+async function clickFilePathOkButton(): Promise<void> {
+  const okButton = await $('*:not([style*="display: none"]).quick-input-action .monaco-button');
```

Comment:
what's `$`?

Reply:
It's a function that's in WDIO (see `node_modules/webdriverio/async.d.ts`).  It's not jQuery, but it has some similarities.